internal/model/command.go

```diff
@@ -193,6 +218,151 @@ func (c ExecutableCommand[F]) GetFlagValues(cmd *cobra.Command) (*F, error) {
 	return &flagValues, nil
 }
 
+// If the command is run from a workflow file, check if the desired version is different from the current version
+// If so, download the desired version and run the command with it as a subprocess
+func runWithVersionFromWorkflowFile(cmd *cobra.Command) error {
+	ctx := cmd.Context()
+	logger := log.From(ctx)
+
+	wf, wfPath, err := utils.GetWorkflow()
+	if err != nil {
+		return fmt.Errorf("failed to load workflow file: %w", err)
+	}
+
+	// If the workflow file doesn't exist, we're running locally, or the --pinned flag is set
+	// simply run the command normally with the existing version of the CLI
+	pinned, _ := cmd.Flags().GetBool("pinned")
+	if wf == nil || pinned || env.IsLocalDev() {
+		return nil
+	}
+
+	currentlyRunningVersion := events.GetSpeakeasyVersionFromContext(ctx)
+	artifactArch := ctx.Value(updates.ArtifactArchContextKey).(string)
+
+	// Try to migrate existing workflows
+	if wf.SpeakeasyVersion == "" {
+		if ghPinned := env.PinnedVersion(); ghPinned != "" {
+			wf.SpeakeasyVersion = workflow.Version(ghPinned)
+		} else {
+			wf.SpeakeasyVersion = "latest"
+		}
+
+		_ = updateWorkflowFile(wf, wfPath)
+	}
+
+	// Get the latest version, or use the pinned version
+	desiredVersion := wf.SpeakeasyVersion.String()
+	if desiredVersion == "latest" {
+		latest, err := updates.GetLatestVersion(artifactArch)
+		if err != nil {
+			return err
+		}
+		desiredVersion = latest.String()
+
+		logger.PrintfStyled(styles.DimmedItalic, "Running with latest Speakeasy version: %s\n", desiredVersion)
+	} else {
+		logger.PrintfStyled(styles.DimmedItalic, "Running with speakeasyVersion from workflow.yaml: %s\n", desiredVersion)
+	}
+
+	// If the desired version is the same as the currently running version of the CLI, just run the command
+	if desiredVersion == currentlyRunningVersion {
+		return nil
```

Comment:
Maybe I'm missing something why do we return nil here

Reply:
It addresses your comment above. If `speakeasyVersion` is set to 1.3.0 and the CLI version executing is already `1.3.0`, then we don't need to download anything and instead just continue normally

Reply:
Keep in mind that this is inside a `preRun`. So returning nil means continue on to the normal `command.Run`